### DISCUSSION TITLE
Arrow keys do not work in item forms

### DISCUSSION THREAD

**Discussion by sxsws** [directus/directus#13989] [Archive]
### Describe the Bug

There's no ability to scroll up and down the form using the arrow keys on the keyboard making the use of a mouse required. Even when setting the focus to the form area.

### To Reproduce

On form.create / .edit click in the form area and press the up/down arrow keys for nothing to happen.

**Comment by benhaynes** [comment]
I'm a bit confused, are you talking about tab indexing through the fields, or advancing on the page using spacebar? Not sure about the arrow keys, unless you mean moving between characters of an input.

**Comment by sxsws** [comment]
You can normally use the arrow keys to move the page up and down

**Comment by benhaynes** [comment]
Which browser/operating system? I'm used to that being the space bar...

**Comment by sxsws** [comment]
Firefox 101, linux. Not even spacebar works for me (but that's down only, not down or up).

**Comment by azrikahar** [comment]
https://user-images.githubusercontent.com/42867097/174523424-0a89b220-50d8-4195-a784-bc43f3a09719.mp4

Can't seem to reproduce this on my end (Windows 10, Chrome), unless this is interface-specific (perhaps certain interface is somehow preventing it? Just a wild guess) or os/browser-specific.

**Reply by sxsws** [reply]
Nope, up nor down arrow certainly doesn't work on firefox ubuntu. Nor does page up/down nor space, nor home/end. "Minimising" the side panel doesn't make any difference. 

I'm not really certain if it is interface related though. It wouldn't make sense if it was! I think the only difference is that I have some half columns and some relations.

And this is an issue, rather than a discussion!

**Reply by azrikahar** [reply]
> And this is an issue, rather than a discussion!

I'm by no means saying it is not possible to be an issue 👍 but we may need to be able to reproduce it reliably before converting it as an issue again. Or if you can share further detail on whether my steps above were not representative of what you tried, it'd be helpful as well.

That said, I am aware that this is very much possible to be a "works on my machine" type of deal, so perhaps @rijkvanzanten can give an additional verification on whether this is is present on his machine. Hope that's understandable.

**Reply by sxsws** [reply]
Try it on Ubuntu Firefox for yourself - a simple vm could be set up quickly from the downloadable iso. It may be the case that selenium tests need to occur on various systems!

**Reply by br41nslug** [reply]
I believe this has to do with which element is focussed, firefox and chrome get different results for me.
In chrome the up/down arrow keys don't work for me on page load until i manually click on the scrollable element.
In firefox clicking does not enable the arrow keys but programmatically focussing the element does (`document.getElementById("main-content").focus()`)
🤔

**Reply by sxsws** [reply]
Yes, you're right it's important where you click to set the focus, but usually, clicking outside form inputs on the parent element usually allows for keyboard scrolling

>  In chrome the up/down arrow keys don't work for me on page load until i manually click on the scrollable element.

This is really unintuitive!

I've since found that if I click in an input and press tab until a checkbox has focus, then the arrow keys work. So there's something on the main content that is preventing keys from working. Anything to do with the CTRL+S etc keyboard capturing code possibly? Maybe it's rejecting all keys and special keys unless the save combos are seen?

It's also the same on listing pages like data model page and results. I have to click on the page background, press tab and then try the arrow keys, and repeat tab/keys again until it works

**Reply by rijkvanzanten** [reply]
I don't think there's anything specific we're doing to prevent that. I think it's caused by the fact that the page content is scrolling in an overflowing element, rather than the page body itself. That would explain why focus needs to be within that context before the browsers default scrolling actions kick in. We're tracking the keyboard a11y improvements over on https://github.com/directus/directus/discussions/2978 👍🏻

**Reply by sxsws** [reply]
But usually, clicking on the page background places the focus in that context so the arrow keys should work?
